resolve: #2 Use pyproject.toml to replace requiremen…

Sorrow-Scarlet opened the pull request:
…ts.txt

1. Add pyproject.toml according to maliang
2. Remove old requirements.txt
3. Successfully tested on Python 3.14 by following commands in venv:
```bash
python3 -m pip install --upgrade build
python3 -m build
```

Refs: #2

Sorrow-Scarlet committed:
build: :construction_worker: use pyproject.toml to replace requirements.txt

1. Add pyproject.toml according to maliang
2. Remove old requirements.txt
3. Successfully tested on Python 3.14 by following commands in venv:
```bash
python3 -m pip install --upgrade build
python3 -m build
```

Refs: #2

Xiaokang2022 commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+maintainers = [{ name = "Xiaokang2022" }, { email = "2951256653@qq.com" }]
+description = "Official designer software of maliang for visualize development"
+readme = "README.md"
+license = { text = "GPL-3.0-only" }
```
这个地方应该直接写 License 的名字就可以了吧？`-only` 有必要加么？我不是很确定，可以查一下

Sorrow-Scarlet replied:
这个地方是依据SPDX的标准，只看到了GPLV3 ONLY和GPLV3 OR LATER，我就暂时选了前者，有需要可以改成后者

Xiaokang2022 commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+dynamic = ["version"]
+requires-python = ">=3.10"
+dependencies = [
+    "maliang>=3.1.5",
```
这个版本的要求是不是太严格了一点？一定要用目前的最新版本及以上版本吗？旧一点的版本应该有些也能运行吧？我不是很确定，可以测试一下 :-)

Sorrow-Scarlet replied:
这个python版本我是根据马良的最低要求来的，至于依赖，貌似写maliang不写版本也可以，但是具体怎么测我还得看看

Xiaokang2022 commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+name = "brush"
+authors = [{ name = "Xiaokang2022" }, { email = "2951256653@qq.com" }]
+maintainers = [{ name = "Xiaokang2022" }, { email = "2951256653@qq.com" }]
+description = "Official designer software of maliang for visualize development"
```
也许我确实是严格了一点，但是，结尾的英文句号不应该被吃掉吧？应该要和 README.md 以及 `__init__.py` 的 docstring 保持一致 :-(

Sorrow-Scarlet replied:
是了，不该被吃，复制粘贴Github的description不仔细导致的，明改改

Sorrow-Scarlet committed:
build(pyproject.toml): Change `description` & `dependencies`

1. Match description according docstring in `README.md` & `__init__.py` .
2. Remove dependencies version to provide wider `maliang` version choice.

NOTE: 移除版本号不影响pyproject.toml使用，在win11下的venv环境中运行下列命令成功安装依赖:
```bash
pip3 install -e .
```
如有需要，可压扁提交为一个（

Xiaokang2022 commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+name = "brush"
+authors = [{ name = "Xiaokang2022" }, { email = "2951256653@qq.com" }]
+maintainers = [{ name = "Xiaokang2022" }, { email = "2951256653@qq.com" }]
+description = "Official designer software of `maliang` for visualize development."
```
```suggestion
description = "Official designer software of maliang for visualize development."
```
这个反引号就不需要了，不是在 README 和 docstring 里面，PyPI 识别不了

Sorrow-Scarlet replied:
长见识了，我没发过PyPI

Xiaokang2022 committed:
Update pyproject.toml
